### PR TITLE
Refactor and clean up actions/worfklows

### PR DESCRIPTION
**Pull request by d3caf**
With the changes we've made to our validation/submission workflows, we no longer need many of the previous actions. This PR removes them and tries to consolidate things a bit as well.

### CODE REVIEW

**Review by mickeyryan42 — APPROVED** [review]
So awesome how many of these we can remove.  as far as i can tell the add-commit-status action isnt used now, is that right?  I dont see a problem leaving it even if so, might be useful sometime later.

File: .github/workflows/validate_packs.yml [hunk 1]
```diff
@@ -1,281 +0,0 @@
...
-  workflow_run:
-    workflows: ["Validation Gate"]
-    types:
-      - completed
-
-env:
-  COMMIT_SHA: ${{ github.event.workflow_run.head_commit.id }}
-
-jobs:
-  ensure-quickstart-ids-are-unique:
```
Comment (josephgregoryii) [inline .github/workflows/validate_packs.yml:13]: I would argue that we would still want this job here. We don't currently have an in-house way to verify this.

File: .github/workflows/validate_packs.yml [hunk 2]
```diff
@@ -1,281 +0,0 @@
...
-        run: cd utils && yarn check-quickstart-uniqueness
-
-      - name: Add commit status
-        if: always()
-        uses: "./.github/actions/add-commit-status"
-        with:
-          statusContext: "Validation / Quickstart id are unique"
-          state: ${{ job.status }}
-
-  ensure-images-are-valid:
```
Comment (josephgregoryii) [inline .github/workflows/validate_packs.yml:48]: Same here

File: .github/workflows/validate_packs.yml [hunk 3]
```diff
@@ -1,281 +0,0 @@
...
-        run: cd utils && yarn validate-images
-
-      - name: Add commit status
-        if: always()
-        uses: "./.github/actions/add-commit-status"
-        with:
-          statusContext: "Validation / Image count and extension compliance"
-          state: ${{ job.status }}
-
-  ensure-icons-exist:
```
Comment (josephgregoryii) [inline .github/workflows/validate_packs.yml:83]: and here

File: .github/workflows/validate_packs.yml [hunk 4]
```diff
@@ -1,281 +0,0 @@
...
-          cd utils && yarn validate-quickstart-data-sources "$URL"
-
-      - name: Add commit status
-        if: always()
-        uses: "./.github/actions/add-commit-status"
-        with:
-          statusContext: "Validation / Data source ids exist"
-          state: ${{ job.status }}
-
-  ensure-quickstart-dashboard-names-are-unique:
```
Comment (josephgregoryii) [inline .github/workflows/validate_packs.yml:208]: here as well

File: .github/workflows/validate_packs.yml [hunk 5]
```diff
@@ -1,281 +0,0 @@
...
-        run: cd utils && yarn check-dashboard-name-uniqueness
-
-      - name: Add commit status
-        if: always()
-        uses: "./.github/actions/add-commit-status"
-        with:
-          statusContext: "Validation / Quickstart dashboard name is unique"
-          state: ${{ job.status }}
-
-  dashboard-helper:
```
Comment (josephgregoryii) [inline .github/workflows/validate_packs.yml:243]: This helps sanitize dashboard JSONs for the catalog service. I think we might still want this as well

**Comment by d3caf** [comment]
@josephgregoryii raining on the deletion parade 😞

**Review by mickeyryan42 — APPROVED** [review]
Thought I approved this yesterday!

**Pull request merged**